src/main/java/uk/gov/justice/hmpps/prison/service/BookingService.java

```diff
@@ -1171,4 +1187,36 @@ public VisitSummary getBookingVisitsSummary(final Long bookingId) {
         final var count = visitInformationRepository.countByBookingId(bookingId);
         return new VisitSummary(visit.map(VisitDetails::getStartTime).orElse(null), count > 0);
     }
+
+    @Transactional
+    @VerifyBookingAccess
+    public PersonalCareNeed addPersonalCareNeed(final Long bookingId, final PersonalCareNeed personalCareNeed) {
+
+        final var nextId = offenderRepository.getNextOffenderHealthProblemId();
+        final var offenderBooking = offenderBookingRepository.findById(bookingId).orElseThrow(EntityNotFoundException.withId(bookingId));
+        final var caseloadType = offenderBooking.getLocation().getType().getCode();
+        final var problemType = healthProblemTypeReferenceCodeRepository.findById(HealthProblemType.pk(personalCareNeed.getProblemType())).orElseThrow(EntityNotFoundException.withId(personalCareNeed.getProblemType()));
+        final var problemCode = healthProblemCodeReferenceCodeRepository.findById(HealthProblemCode.pk(personalCareNeed.getProblemCode())).orElseThrow(EntityNotFoundException.withId(personalCareNeed.getProblemCode()));
+        final var problemStatus = healthProblemStatusReferenceCodeRepository.findById(HealthProblemStatus.pk(personalCareNeed.getProblemStatus())).orElseThrow(EntityNotFoundException.withId(personalCareNeed.getProblemStatus()));
+
+        if(!problemCode.hasParent(problemType)){
+                throw BadRequestException.withMessage("Problem code has incorrect type");
+        }
+
+        final var offenderHealthProblem = OffenderHealthProblem.builder()
+            .id(nextId)
+            .offenderBooking(offenderBooking)
+            .caseloadType(caseloadType)
+            .problemType(problemType)
+            .problemCode(problemCode)
+            .problemStatus(problemStatus)
+            .description(personalCareNeed.getProblemDescription())
+            .startDate(personalCareNeed.getStartDate())
+            .endDate(personalCareNeed.getEndDate())
+            .build();
+
+        offenderBooking.add(offenderHealthProblem);
+
+        return personalCareNeed;
```

Comment:
To get ID - I would suggest loading outside the transaction

Reply:
we no longer need to return any details so following iep example and returning nil response